server/src/main/scala/org/http4s/server/middleware/Jsonp.scala

```diff
@@ -0,0 +1,64 @@
+
+  def apply(callbackParam: String)(service: HttpService): HttpService =
+    HttpService.lift { req =>
+      req.params.get(callbackParam) match {
+        case Some(ValidCallback(callback)) =>
+          service(req).map { resp =>
+            resp.contentType.map(_.mediaType) match {
+              case Some(MediaType.`application/json`) =>
+                jsonp(resp, callback)
+              case _ =>
+                resp
+            }
+          }
+        case Some(invalidCallback) =>
+          logger.warn(s"Jsonp requested with invalid callback function name $invalidCallback")
+          Response(Status.BadRequest).
+            withBody(s"Not a valid callback name: $invalidCallback")
```

Comment:
We generally don't reflect anything back from the request in error messages to avoid any XSS vectors.  Being `text/plain` helps a lot, but is not foolproof.  Maybe just `"Not a valid callback name"?